p2p-tests/maintainer-update-propagation-3-test.ts

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env -S node --require ts-node/register/transpile-only --require tsconfig-paths/register
+
+// Copyright © 2021 The Radicle Upstream Contributors
+//
+// This file is part of radicle-upstream, distributed under the GPLv3
+// with Radicle Linking Exception. For full terms see the included
+// LICENSE file.
+
+import * as path from "path";
+import { strict as strictAssert } from "assert";
+
+import * as source from "proxy-client/source";
+import { sleep } from "ui/src/sleep";
+import {
+  RadicleProxy,
+  UpstreamSeed,
+  commit,
+  getLatestCommitSha,
+  pushRad,
+  radCli,
+  runTestcase,
+  withRetry,
+} from "./lib/p2p";
+
+async function testcase(dataPath: string) {
+  const project = {
+    name: "my-fancy-project",
+    urn: "rad:git:hnrkrhhs1goaawo7db1gpyct8hd7mif5q8c3o",
+  };
+
+  const seed = new UpstreamSeed({
+    namespace: "seed",
+    ipAddress: "10.0.0.1",
+    project: project.urn,
+    dataPath,
+  });
+
+  const maintainer = new RadicleProxy({
+    namespace: "maintainer",
+    ipAddress: "10.0.0.101",
+    seed: seed.seedAddress,
+    dataPath,
+  });
+  const maintainerProjectCheckoutPath = path.join(
+    maintainer.checkoutPath,
+    project.name
+  );
+
+  const contributor = new RadicleProxy({
+    namespace: "contributor",
+    ipAddress: "10.0.0.102",
+    seed: seed.seedAddress,
+    dataPath,
+  });
+
+  seed.start();
+  maintainer.start();
+  contributor.start();
+
+  // Without this the test fails, not sure why.
```

Comment:
Can we note this down in an issue? This might indicate that something is bugged and it’s great to have a failing test case

Reply:
https://github.com/radicle-dev/radicle-upstream/issues/2669